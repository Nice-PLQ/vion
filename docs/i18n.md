# 国际化

Vion 默认采用中文语言，如果需要切换其他语言，请参考如下

### 1、切换语言

Vion 通过 Locale 组件实现多语言支持，使用`Locale.use`方法切换语言环境，比如切换到英文环境

```js
// 按需引入
import { createApp } from "vue";
import { ActionSheet, Locale } from "vion";
import messages from "vion/es/locale/lang/en-US";

Locale.use("en-US", messages);

createApp()
  .use(ActionSheet);
  .use(Locale);

```

或

```js
// 全量引入Vion
import { createApp } from "vue";
import Vion from "vion";
import messages from "vion/es/locale/lang/en-US";

vion.Locale.use("en-US", messages);

createApp().use(Vion);
```

### 2、语言包

目前 vion 内置了以下几种语言

- 简体中文（zh-CN）
- 繁体中文（zh-Hant）
- 英语（en-US）
- 日语（ja-JP）

如果不能满足业务场景，只需要按照[语言包格式](https://github.com/Nice-PLQ/vion/blob/master/src/locale/lang/zh-CN.js)自定义语言，然后使用`Locale`组件加载即可，如：

```js
import messages from "./lang/fr-FR"; // 自定义的【法语】语言包
Locale.use("fr-FR", messages);
```

### 3、引入多个语言包

```js
import frMessages from "./lang/fr-FR"; // 自定义的【法语】语言包
import trMessages from "./lang/tr-TR"; // 自定义的【土耳其语】语言包

Locale.add("tr-TR", trMessages); // 将语言包加入到集合中
Locale.use("fr-FR", frMessages); // 默认使用法语作为首选语言
```
