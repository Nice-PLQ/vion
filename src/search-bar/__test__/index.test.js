import { mount } from '@vue/test-utils';
import { next } from '../../../test';
import SearchBar from '..';
import i18n from '../../locale';

describe('SearchBar', () => {
  const wrapper = mount(SearchBar, {
    props: {
      type: 'capsule',
      cancelBtnVisible: true,
    },
    global: {
      config: {
        globalProperties: {
          $locale: i18n
        }
      }
    }
  });
  test('渲染', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('focus & blur事件触发', async () => {
    wrapper.find('.v-search-bar_unfocus').trigger('click');
    await next();
    expect(wrapper.element).toMatchSnapshot();

    const input = wrapper.find('input');
    expect(input.exists()).toBe(true);
    expect(wrapper.emitted('focus')).toBeTruthy();
    input.trigger('blur');
    expect(wrapper.emitted('blur')).toBeTruthy();
  });

  test('change & search事件触发', async () => {
    wrapper.find('.v-search-bar_unfocus').trigger('click');
    await next();
    const input = wrapper.find('input');
    input.element.value = 'test';
    input.trigger('input');
    expect(wrapper.emitted('change')[0][0]).toBe('test');

    input.trigger('keyup.enter');
    expect(wrapper.emitted('search')[0][0]).toBe('test');
  });

  test('clear & back-btn-click事件触发', async () => {
    const wrapper = mount(SearchBar, {
      props: {
        backBtnVisible: true,
        value: 'test'
      },
      global: {
        config: {
          globalProperties: {
            $locale: i18n
          }
        }
      }
    });
    wrapper.find('.v-search-bar_focus input').trigger('click');
    await next();
    wrapper.find('.v-search-bar__clear-icon').trigger('click');
    wrapper.find('.v-search-bar__back-icon').trigger('click');
    expect(wrapper.emitted('clear')).toBeTruthy();
    expect(wrapper.emitted('back-btn-click')).toBeTruthy();
  });

  test('cancel事件触发', async () => {
    const wrapper = mount(SearchBar, {
      props: {
        cancelBtnVisible: true,
        value: 'test'
      },
      global: {
        config: {
          globalProperties: {
            $locale: i18n,
          }
        }
      }
    });
    wrapper.find('.v-search-bar_unfocus').trigger('click');
    await next();
    wrapper.find('.v-search-bar__cancel').trigger('click');
    expect(wrapper.emitted('cancel')).toBeTruthy();
  });
});
