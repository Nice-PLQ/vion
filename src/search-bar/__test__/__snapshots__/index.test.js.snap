// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar focus & blur事件触发 1`] = `
<div
  class="v-search-bar"
>
  <!--v-if-->
  <form
    action=""
    class="v-search-bar__capsule"
  >
    <div
      class="v-search-bar_focus"
    >
      <i
        class="v-icon v-search-bar__search-icon"
        style="display: inline-block; width: 1em; height: 1em; line-height: 1em;"
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8 20C8 13.3726 13.3726 8 20 8C26.6274 8 32 13.3726 32 20C32 26.6274 26.6274 32 20 32C13.3726 32 8 26.6274 8 20ZM20 4C11.1634 4 4 11.1634 4 20C4 28.8366 11.1634 36 20 36C23.7514 36 27.2013 34.7089 29.9295 32.5469L40.9371 43.6652L43.7796 40.8509L32.7317 29.6918C34.7823 27.0022 36 23.6433 36 20C36 11.1634 28.8366 4 20 4Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </i>
      <input
        placeholder="搜索"
        type="search"
      />
      <!--v-if-->
    </div>
  </form>
  <div
    class="v-search-bar__cancel"
  >
    取消
  </div>
</div>
`;

exports[`SearchBar 渲染 1`] = `
<div
  class="v-search-bar"
>
  <!--v-if-->
  <form
    action=""
    class="v-search-bar__capsule"
  >
    <div
      class="v-search-bar_unfocus"
    >
      <i
        class="v-icon v-search-bar_search-icon_unfocus"
        style="display: inline-block; width: 1em; height: 1em; line-height: 1em;"
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8 20C8 13.3726 13.3726 8 20 8C26.6274 8 32 13.3726 32 20C32 26.6274 26.6274 32 20 32C13.3726 32 8 26.6274 8 20ZM20 4C11.1634 4 4 11.1634 4 20C4 28.8366 11.1634 36 20 36C23.7514 36 27.2013 34.7089 29.9295 32.5469L40.9371 43.6652L43.7796 40.8509L32.7317 29.6918C34.7823 27.0022 36 23.6433 36 20C36 11.1634 28.8366 4 20 4Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </i>
      <p>
        搜索
      </p>
    </div>
  </form>
  <!--v-if-->
</div>
`;
